streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -588,12 +588,16 @@ class StreamingContext private[streaming] (
     state match {
       case INITIALIZED =>
         startSite.set(DStream.getCreationSite())
-        sparkContext.setCallSite(startSite.get)
         StreamingContext.ACTIVATION_LOCK.synchronized {
           StreamingContext.assertNoOtherContextIsActive()
           try {
             validate()
-            scheduler.start()
+            ThreadUtils.runInNewThread("streaming-start") {
+              sparkContext.setCallSite(startSite.get)
```

Comment:
an *inherited* thread local variable

(anyway we can just keep this change, not a big deal)